pinot-query-runtime/src/main/java/org/apache/pinot/query/service/SubmissionService.java

```diff
@@ -24,29 +24,30 @@
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Future;
 import java.util.concurrent.TimeUnit;
+import org.apache.pinot.common.exception.QueryException;
 
 
 /**
  * Submission service is used to submit multiple runnables and checks the result upon all {@link Future} returns
  * or any failure occurs.
  */
 public class SubmissionService {
-  private final ExecutorService _executor;
   private final List<CompletableFuture<Void>> _futures = new ArrayList<>();
 
-  public SubmissionService(ExecutorService executor) {
-    _executor = executor;
+  public SubmissionService() {
```

Comment:
This is no longer a service. IMO it might be more readable if we directly implement this logic in `QueryServer.submit()`. I don't really see the value of this wrapper class

Reply:
good point. will convert it into a util --> its only job is to maintain the list of futures for completion